src/libraries/System.Text.Json/src/System/Text/Json/Serialization/PooledByteBufferWriter.cs

```diff
@@ -12,7 +12,7 @@ namespace System.Text.Json
 {
     internal sealed class PooledByteBufferWriter : IBufferWriter<byte>, IDisposable
     {
-        private byte[] _rentedBuffer;
+        private byte[]? _rentedBuffer;
```

Comment:
Given `_rentedBuffer` will only ever be null after `Dispose()`, should we keep this as non-nullable and bang the `_rentedBuffer = null!;` in `Dispose()`?

Reply:
We can, seems always initi-ed, somehow all accesses are asserting for null before each use. Anyways I can remove this `?` and add '!' on Dispose